### PR TITLE
PYT-1322 Add xpath-injection triggers to vulnpy

### PR DESCRIPTION
**Pull request by evtimov9**

### CODE REVIEW

File: src/vulnpy/trigger/xpath.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+def do_lxml_etree_findall(user_input):
+
+    tree = lxml.etree.parse(StringIO(xml))
+
+    tree.xpath('/planet[name[text()="Kepler"]]')
+    root = tree.getroot()
+    root.findall(user_input)
+
+    return 200
```
Comment (lazorchakp) [inline src/vulnpy/trigger/xpath.py:28]: these aren't really view functions, and are intended to be useful outside of the context of a web application. Instead of returning 200, can you return a result relevant to the trigger function?

For example, you could return a stringified version of the return value from `root.findall`. Same for all functions here

File: src/vulnpy/trigger/xpath.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import lxml.etree
+import xml.etree.ElementTree as ET
+from io import StringIO
+
+xml = (
```
Comment (lazorchakp) [inline src/vulnpy/trigger/xpath.py:5]: could you make this a triple quoted string? it would be easier to read.

Also, it looks like py27 tests are failing because this must always be unicode. Changing it to
```
u"""
......
"""
```
should work

File: src/vulnpy/trigger/xpath.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+</planet>"
+    "</planets>"
+)
+
+
+def do_lxml_etree_findall(user_input):
+
+    tree = lxml.etree.parse(StringIO(xml))
+
+    tree.xpath('/planet[name[text()="Kepler"]]')
```
Comment (lazorchakp) [inline src/vulnpy/trigger/xpath.py:24]: so admittedly I'm not too familiar with xml parsing or this vulnerability type. Would it make sense to move the initial parsing and call to `xpath` to the module level instead of inside of each function?

**Review by lazorchakp — CHANGES_REQUESTED** [review]

**Review by lazorchakp — APPROVED** [review]
looks good to me - if you end up needing to iterate a little in order to get framework tests passing, no worries. It happens more often than you'd think 🙂
